frontend/src/lib/components/TaxonomicFilter/infiniteListLogic.ts

```diff
@@ -247,18 +247,11 @@ export const infiniteListLogic = kea<infiniteListLogicType>([
 
                     if (group?.logic && group?.value) {
                         let items = group.logic.selectors[group.value]?.(state)
+
                         // Handle paginated responses for cohorts, which return a CountedPaginatedResponse
                         if (items?.results) {
                             items = items.results
                         }
-                        // TRICKY: Feature flags don't support dynamic behavioral cohorts,
-                        // so we don't want to show them as selectable options in the taxonomic filter
-                        // in the feature flag UI.
-                        // TODO: Once we support dynamic behavioral cohorts, we should show them in the taxonomic filter,
-                        // and remove this kludge.
-                        if (Array.isArray(items) && items.every((item) => 'filters' in item)) {
-                            return filterOutBehavioralCohorts(items, props.hideBehavioralCohorts)
-                        }
```

Comment:
don't filter on the client-side because it could make the counts look funny: e.g. if 35 of the first 100 cohorts fetched were behavioral, it would show 65 cohorts in the filter list, but say we have 150 cohorts, and only 35 are behavioral, I'd rather show 100 in the tag than 65).